table/block_based/block_like_traits.h

```diff
@@ -32,14 +55,32 @@ class BlocklikeTraits<BlockContents> {
     return 0;
   }
 
-  static Cache::DeleterFn GetDeleter(BlockType block_type) {
+  static size_t SizeCallback(void* obj) {
+    assert(obj != nullptr);
+    BlockContents* ptr = reinterpret_cast<BlockContents*>(obj);
+    return ptr->data.size();
+  }
+
+  static Status SaveToCallback(void* obj, size_t offset, size_t size,
```

Comment:
Surely we can share most of the code between the SaveToCallback implementations. In fact, the appropriate "trait" seems to be getting a Slice of underlying data from a `void* obj`. From that trait, you can have unified (template) implementations of SizeCallback and SaveToCallback.

Reply:
One case I'm not sure is ParsedFullFilterBlock. Its data is in the block_content_ as a private member